packages/calypso-e2e/src/element-helper.ts

```diff
@@ -75,3 +75,30 @@ export async function clickNavTab( page: Page, name: string ): Promise< void > {
 	const elementHandle = await page.waitForSelector( selectors.navTab( name ) );
 	await Promise.all( [ page.waitForNavigation(), elementHandle.click() ] );
 }
+
+/**
+ * Retry any action up to three times or action passes without throwing an exception,
+ * whichever comes first.
+ *
+ * This is useful for situations where the backend must process the results of a
+ * previous action then inform the front end of the result of the process.
+ * An example of this is the user invitation system where the backend must receive,
+ * verify and send the invitation issued by a user. If the resulting checks were
+ * successful, the resulting invitation is shown as 'pending'.
+ *
+ * @param {Page} page Page object.
+ */
+export async function retryAction(
```

Comment:
Do you think the majority use case will be waiting for a selector and then reloading? If so, what do you think of adding a second library function that takes a page and selector and handles creating the waiting function? Again, totally optional :)

Reply:
I agree that majority of the use case would be to wait on a single action/selector.
With that said, I would prefer to keep the current signature for the time being - one less complication and hopefully POM writers are experienced enough to know what a closure is :)